src/ipahealthcheck/ipa/certs.py

```diff
@@ -583,6 +584,103 @@ def check(self):
                     'verifying trust')
 
 
+@registry
+class IPACertMatchCheck(IPAPlugin):
+    """ Ensure certificates match between LDAP and NSS databases """
+
+    requires = ('dirsrv',)
+
+    @duration
+    def check(self):
+
+        # Ensure certificates present in LDAP and NSSDB match
+        def match_ldap_nss_cert(plugin, ldap, db, cert_dn, attr, cert_nick):
+            try:
+                entry = ldap.get_entry(cert_dn)
+            except errors.NotFound:
+                yield Result(plugin, constants.ERROR,
+                             msg='%s entry not found in LDAP' % cert_dn)
+                return False
+            try:
+                nsscert = db.get_cert_from_db(cert_nick)
+            except Exception as e:
+                yield Result(plugin, constants.ERROR,
+                             error=str(e),
+                             msg=('Unable to load %s certificate:'
+                                  '{error}' % cert_nick))
+                return False
+            ldapcert = entry[attr][0]
+            if nsscert != ldapcert:
+                yield Result(plugin, constants.ERROR,
+                             msg=('\'%s\' certificate in NSS DB does not '
+                                  'does not match entry in LDAP' % cert_nick))
+                return False
+            return True
+
+        # Ensure /etc/ipa/ca.crt matches NSS DB CA certificates
+        def match_cacert_and_db(plugin, cacert, dbpath):
+            db = certs.CertDB(api.env.realm, dbpath)
+            try:
+                dbcacert = db.get_cert_from_db('%s IPA CA' % api.env.realm)
+            except Exception as e:
+                yield Result(plugin, constants.ERROR,
+                             error=str(e),
+                             msg='Unable to load CA cert: {error}')
+                return False
+            if cacert != dbcacert:
+                yield Result(plugin, constants.ERROR,
+                             msg=('CA Certificate from %s does not match %s'
+                                  % (dbpath, paths.IPA_CA_CRT)))
+                return False
+            return True
+
+        db = certs.CertDB(api.env.realm, paths.PKI_TOMCAT_ALIAS_DIR)
+        dn = DN('uid=pkidbuser,ou=people,o=ipaca')
+        nick = 'subsystemCert cert-pki-ca'
+        subsystem_ok = yield from match_ldap_nss_cert(self, self.conn,
+                                                      db, dn,
+                                                      'userCertificate', nick)
```

Comment:
Arguably this check could be in src/ipahealthcheck/dogtag/ca.py since it's a dogtag cert.

Be sure to test this in a ca-less environment as well.